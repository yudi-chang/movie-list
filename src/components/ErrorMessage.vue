<template>
  <div class="error-message">
    <img src="/images/error-logo.png" alt="Unexpected error image" />
    <h2 class="mb-12 text-xl fw-bold">Oops something went wrong</h2>
    <div class="message">An unexpected error has occured. please try again or contact support if the error persists</div>
    <button @click="retry()" class="mt-20">
      Try again
    </button>
  </div>
</template>

<script setup lang="ts">
const emit = defineEmits<{
  (event: 'retry-fetch'): void;
}>()

function retry(): void {
  emit('retry-fetch');
}
</script>

<style scoped lang="scss">
.error-message {
  text-align: center;
  width: 100%;
  padding: 100px 20px 0 20px;

  img {
    max-width: 100%;
    margin: 0 auto;
  }
}

.message {
  width: 500px;
  margin: 0 auto;
  max-width: 100%;
  padding: 0 20px;
}

button {
  width: 250px;
  border-radius: 5px;
  background-color: #1289A7;
  color:$secondary-color;
  padding: 10px 20px;
  max-width: 100%;
  cursor: pointer;
  border: none;
  font-size: 17px;
}
</style>
