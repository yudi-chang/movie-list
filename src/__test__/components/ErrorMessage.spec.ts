import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import ErrorMessage from '@/components/ErrorMessage.vue'

describe('ErrorMessage', () => {
  const defaultParams = {};

  const createWrapper = (params = defaultParams) => {
    return mount(ErrorMessage, params)
  }

  it('should renders and have button click working properly (emit retry-fetch)', async () => {
    const wrapper = createWrapper();
    
    const errorMessage = wrapper.find('[data-test="error-message-wrapper"]');
    expect(errorMessage).toMatchSnapshot();

    await wrapper.find('[data-test="retry-btn"]').trigger('click');
    expect(wrapper.emitted('retry-fetch')).toBeTruthy();
    expect(wrapper.emitted('retry-fetch')?.length).toBe(1);
  })
})
