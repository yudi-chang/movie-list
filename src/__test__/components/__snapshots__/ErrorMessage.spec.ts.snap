// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ErrorMessage > should renders and have button click working properly (emit retry-fetch) 1`] = `
DOMWrapper {
  "isDisabled": [Function],
  "wrapperElement": <div
    class="error-message"
    data-test="error-message-wrapper"
    data-v-54924f92=""
  >
    <img
      alt="Unexpected error image"
      data-v-54924f92=""
      src="/images/error-logo.png"
    />
    <h2
      class="mb-12 text-xl fw-bold"
      data-v-54924f92=""
    >
      Oops something went wrong
    </h2>
    <div
      class="message"
      data-v-54924f92=""
    >
      An unexpected error has occured. please try again or contact support if the error persists
    </div>
    <button
      class="mt-20"
      data-test="retry-btn"
      data-v-54924f92=""
    >
       Try again 
    </button>
  </div>,
}
`;
